components/makeSubmit.tsx

```diff
@@ -17,33 +17,47 @@ export const makeSubmit = (
   const buttonProps = Array.isArray(props) ? props : [props];
 
   const Submit = ({ onSubmit }: SubmitProps): React.ReactElement => {
+    const router = useRouter();
+    const buttonUrlsString = buttonProps
+      .map(({ url }) => url.pathname)
+      .join(",");
+
+    useEffect(() => {
+      buttonProps.map(({ url }) => {
+        const { href } = urlsForRouter(url);
+
+        router.prefetch(href.pathname);
+      });
+    }, [router, buttonUrlsString]);
+
     return (
       <>
         {buttonProps.map(({ url, value }, i) => {
           const { href, as } = urlsForRouter(url);
 
           return (
-            <NextLink key={i} href={href} as={as}>
-              <Button
-                className={
-                  i > 0
-                    ? "submit-button lbh-button--secondary govuk-button--secondary"
-                    : "submit-button"
+            <Button
+              key={as.pathname}
+              className={
+                i > 0
+                  ? "submit-button lbh-button--secondary govuk-button--secondary"
+                  : "submit-button"
+              }
+              onClick={async (): Promise<void> => {
+                try {
+                  await onSubmit();
+                } catch (error) {
+                  // This is invisible to the user, so we should do something to
+                  // display it to them.
```

Comment:
Has this been captured somewhere?

Reply:
There's a backlog task for making sure we handle errors well generally.